src/org/labkey/targetedms/query/PeptideManager.java

```diff
@@ -115,10 +120,68 @@ public static List<PeptideCharacteristic> getPeptideCharacteristic(long peptideG
         sql.append(" ON gm.id = pep.Id");
         sql.append(" INNER JOIN ").append(TargetedMSManager.getTableInfoSampleFile(), "sf");
         sql.append(" ON sf.Id = pci.SampleFileId");
+        if (!Objects.equals(replicateId, Long.valueOf(0)))
+        {
+            sql.append(" INNER JOIN ").append(TargetedMSManager.getTableInfoReplicate(), "rep");
+            sql.append(" ON rep.Id = sf.ReplicateId");
+        }
+        sql.append(" WHERE gm.PeptideGroupId=? ");
+        if (!Objects.equals(replicateId, Long.valueOf(0)))
+        {
+            sql.append(" AND rep.Id=? ");
+        }
+        sql.append(" GROUP BY pep.Sequence,pci.SampleFileId, pep.StartIndex, pep.EndIndex ) X ");
+        sql.append(" GROUP BY X.Sequence, X.StartIndex, X.EndIndex ");
+        sql.add(peptideGroupId);
+        if (!Objects.equals(replicateId, Long.valueOf(0)))
+        {
+            sql.add(replicateId);
+        }
+        return new SqlSelector(TargetedMSManager.getSchema(), sql).getArrayList(PeptideCharacteristic.class);
+    }
+
+    public static List<PeptideCharacteristic> getModifiedPeptideCharacteristics(long peptideGroupId, @Nullable Long replicateId)
+    {
+        var sqlDialect = TargetedMSManager.getSqlDialect();
+        var pgLog10Intensity = "LOG(" + sqlDialect.getNumericCast(new SQLFragment("MAX(X.Intensity)")).getSQL() + ")";
```

Comment:
Can we consolidate at least the PG/SQLServer special casing with the previous method's? Maybe factor it out to an `appendLogExpressions()` method that takes the SqlDialect and appends the right variants?